scripts/deploy.py

```diff
@@ -0,0 +1,352 @@
+import os
+import sys
+import logging
+import boto3
+import botocore.exceptions
+import json
+import requests
+
+logging.basicConfig()
+logging.captureWarnings(True)
+
+
+class DeployIRCHooky(object):
+
+    def __init__(self):
+        self.log = logging.getLogger("irchooky")
+        self.log.setLevel(logging.DEBUG)
+
+    def validate_env_vars(self):
+        env_vars = [
+            'AWS_ACCESS_KEY_ID',
+            'AWS_SECRET_ACCESS_KEY',
+            'AWS_DEFAULT_REGION',
+            'LAMBDA_FUNCTION_NAME',
+            'REST_ENDPOINT_NAME',
+            'IRCHOOKY_IRC_SERVER',
+            'IRCHOOKY_IRC_PORT',
+            'IRCHOOKY_IRC_CHANNEL'
+        ]
+        for var in env_vars:
+            if not os.environ.get(var):
+                self.log.error("Could not find environment variable %s" % var)
+                sys.exit(1)
+        self.lambda_func_name = os.environ['LAMBDA_FUNCTION_NAME']
+        self.rest_endpoint_name = os.environ['REST_ENDPOINT_NAME']
+        self.region = os.environ['AWS_DEFAULT_REGION']
+        self.irc_server = os.environ['IRCHOOKY_IRC_SERVER']
+        self.irc_port = os.environ['IRCHOOKY_IRC_PORT']
+        self.irc_channel = os.environ['IRCHOOKY_IRC_CHANNEL']
+
+    def create_lambda_policy(self):
+        policy_name = "%s-lambda-basic-execution-role-policy" % self.lambda_func_name  # NOQA
+        policy_doc = {
+            "Version": "2012-10-17",
+            "Statement": [
+                {
+                    "Effect": "Allow",
+                    "Action": [
+                        "logs:CreateLogGroup",
+                        "logs:CreateLogStream",
+                        "logs:PutLogEvents",
+                        "sns:Publish",
+                        "sns:Subscribe"
+                    ],
+                    "Resource": [
+                        "arn:aws:logs:*:*:*",
+                        "arn:aws:sns:*"
+                    ]
+                }
+            ]
+        }
+        client = boto3.client('iam')
+        self.lambda_policy_arn = ""
+        try:
+            response = client.create_policy(
+                PolicyName=policy_name,
+                PolicyDocument=json.dumps(policy_doc)
+            )
+            self.lambda_policy_arn = response['Policy']['Arn']
+            self.log.info("Policy %s successfully created." % self.lambda_policy_arn)  # NOQA
+        except botocore.exceptions.ClientError:
```

Comment:
I have found boto's exception bubbling to be not so great :disappointed: 

You bring up a good point and I'm going to keep this in mind.

(thank you for reading this PR :smile:)